tir/tirClient.go

```diff
@@ -80,12 +83,20 @@ func NewTirHttpClient(tokenProvider TokenProvider, config config.M2M) (client Ti
 	_, err = httpcache.NewWithInmemoryCache(httpClient, true, time.Second*60)
 	if err != nil {
 		logging.Log().Errorf("Was not able to inject the cache to the client. Err: %v", err)
-		return
+		return nil, err
```

Comment:
I know its nitpicking but the returned err will (hopefully) be logged further up in the calling code, so if we log here too, we will get two log entries reducing the readability